components/locid/src/extensions/mod.rs

```diff
@@ -61,6 +61,7 @@ use crate::parser::ParserError;
 use crate::parser::SubtagIterator;
 /// Defines the type of extension.
 #[derive(Debug, PartialEq, Eq, Clone, Hash, PartialOrd, Ord, Copy)]
+#[allow(clippy::exhaustive_enums)] // this type is stable
```

Comment:
Nit: Probably `non_exhaustive`. People should not be matching on this enum, and code that uses it should generally be robust. I chatted with Zibi about this last week and we have other methods like `.retain_type()` to handle this situation.